packages/components/src/components/hds/app-side-nav/list/title.hbs

```diff
@@ -4,8 +4,8 @@
 }}
 
 <Hds::AppSideNav::List::Item>
-  <div
+  <h3
     class="hds-app-side-nav__list-title hds-typography-body-100 hds-font-weight-semibold"
     ...attributes
-  >{{~yield~}}</div>
+  >{{~yield~}}</h3>
```

Comment:
I change the div tags to h3 tags for now.

Reply:
I'm not sure how I would be able to associate the ul with the title using aria-labelled by with the current structure. I think the whole structure would have to be redone to make it work. 

(I worked on the SideNav in the early stages and wanted to structure it as headers outside of uls which served as the title of the uls but was overruled in favor of the current structure. The structure has complexities due to some of how it functions within Products I believe.)

Reply:
If we really need to do this (aria-labelled changes to associate the Title headers with the Lists) then I could make another branch of the component PR as it's going to require more extensive changes and testing than the other smaller updates I've made in this current branch. Please let me know what you think.

Reply:
1. The lists do need to have accNames or else they just show up as "list" in the rotor
2. I think it would be sufficient to have this list item given an `id` attribute and then use `aria-labelledby` on the `ul`. 

I don't think these should be h3's.

Reply:
Can you take a look at the code and let me know if you have ideas on how to do it with the current structure? Redoing the current structure if something I really want to avoid unless absolutely necessary.

If the current structure is unacceptable on the AppSideNav it will need to be updated on the SideNav too.

@shleewhite @alex-ju Please let me know if you have any input on this.

Reply:
Hmmm.. I hadn't realized before that it is just one list with everything inside it... this is tough because visually theyre either separate lists or sub-lists, both of which would require refactoring parts of the structure. 

Can you clarify why you would need to update it in sidenav too? Isn't it going to be deprecated?

Reply:
Update: From squad discussions, we will avoid making new updates the the SideNav unless absolutely necessary.

Reply:
Let me know if you have ideas on improvements that could be made to the current structure or if the updates I already made are ok for now.

Reply:
I pushed up some changes to add ids to Titles and associate them with the `ul`.

Reply:
After reviewing it again, I think we should just take the work for labelling the lists by the titles out of scope for now. The reason being, now there is still one large list called "Services Default Org". 

We need to come back to this and restructure the components slightly as demonstrated in this spike: https://github.com/hashicorp/design-system/pull/2478/files

Unfortunately, I don't think there are any small wins possible for the sublists/titles without changing the structure.

I can approve this PR once it is scoped down to just adding the visually hidden H2 and labelling the nav by the H2.

Reply:
Thanks @shleewhite 

@MelSumner could you please let me know your input? Are you ok with what Lee said above? I'd like to know any specific changes I should make to this current PR so I can get it merged into the component branch (if you would like changes other than what Lee recommended).

Thanks!

Reply:
For now, I changed the `h3` I had added wrapping the Title content back into a `div`